File: src/main/java/brightspark/landmanager/block/LMBlocks.java
package brightspark.landmanager.block;

import brightspark.landmanager.LandManager;
import net.minecraft.block.Block;
import net.minecraft.client.renderer.block.model.ModelResourceLocation;
import net.minecraft.item.Item;
import net.minecraft.item.ItemBlock;
import net.minecraftforge.client.event.ModelRegistryEvent;
import net.minecraftforge.client.model.ModelLoader;
import net.minecraftforge.event.RegistryEvent;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

@Mod.EventBusSubscriber(modid = LandManager.MOD_ID)
public class LMBlocks
{
	public static final Block home = new BlockHome();

	@SubscribeEvent
	public static void regBlocks(RegistryEvent.Register<Block> event)
	{
		event.getRegistry().register(home);
	}

	@SubscribeEvent
	public static void regItems(RegistryEvent.Register<Item> event)
	{
		event.getRegistry().register(new ItemBlock(home).setRegistryName(home.getRegistryName()));
	}

	@SideOnly(Side.CLIENT)
	@SubscribeEvent
	public static void regModels(ModelRegistryEvent event)
	{
		ModelLoader.setCustomModelResourceLocation(Item.getItemFromBlock(home), 0, new ModelResourceLocation(home.getRegistryName(), "inventory"));
	}
}

File: src/main/kotlin/brightspark/landmanager/init/LMBlocks.kt
package brightspark.landmanager.init

import brightspark.landmanager.block.HomeBlock
import brightspark.landmanager.util.setRegName
import net.minecraft.block.AbstractBlock
import net.minecraft.block.Block
import net.minecraft.block.material.Material
import net.minecraftforge.event.RegistryEvent
import thedarkcolour.kotlinforforge.forge.objectHolder

object LMBlocks {
	val HOME: Block by objectHolder("home")

	fun register(event: RegistryEvent.Register<Block>) = event.registry.register(
		HomeBlock(props(Material.WOOD)).setRegName("home")
	)

	private fun props(material: Material): AbstractBlock.Properties = AbstractBlock.Properties.create(material)
}
